### PR TITLE
New images for 'Message Log'

### PR DESCRIPTION
**Pull request by GScottElshaw**
With #2764 landing, some additional colour to help differentiate the different log types.

### CODE REVIEW

**Comment by ckerr** [comment]
I don't really know good macOS UI so I'd rather defer to another reviewer, but this PR's been green for a week, so:

unless someone requests changes / is against this PR, I'm going to merge it tomorrow :smiley_cat: @livings124 @mikedld

**Comment by GScottElshaw** [comment]
> I don't really know good macOS UI so I'd rather defer to another reviewer, but this PR's been green for a week, so:
> 
> unless someone requests changes / is against this PR, I'm going to merge it tomorrow 😺 

@ckerr Can Mr @DevilDimon possibly have a look?

**Comment by DevilDimon** [comment]
The code is in macosx/MessageWindowController.mm, you'll need to create the right NSImages within `tableView:objectValueForTableColumn:` method. You might take a look how other images are created there for inspiration

**Comment by GScottElshaw** [comment]
> You might take a look how other images are created there for inspiration

That's certainly how it worked for this PR 👍🏼  Thanks for the directions @DevilDimon.

**Review by mikedld — CHANGES_REQUESTED** [review]
1. Comparing existing and new icons, the former are circles with a slightly darker outline while the latter are octagons:

   image

   I think we need to take the new ones from the same place the old ones came from, which may well be macOS itself, looking at its general preferences dialog (although I see that the colors are in fact slightly different, at least in 12.2):

   image

2. New icons aren't size-optimized which I think we did at some point because the size difference in dramatic, e.g. for 2x variants, existing ones are around 275 bytes while new ones are around 2300-2600 bytes.

3. New icons need to be added to macosx/CMakeLists.txt: https://github.com/transmission/transmission/blob/bd895b50526665c4c379a5670105cafe017d2ad1/macosx/CMakeLists.txt#L238

File: .github/workflows/objective-c-xcode.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Xcode - Build and Analyze
```
[the remark this answers is not in the document]
Comment (GScottElshaw) [reply .github/workflows/objective-c-xcode.yml:1]: > We're not using GitHub Actions, and I'm not really sure what this checks and why we need it.

I was using GitHub Desktop for this particular PR.

**Comment by GScottElshaw** [comment]
> 1. Comparing existing and new icons, the former are circles with a slightly darker outline while the latter are octagons:
>    image
>    I think we need to take the new ones from the same place the old ones came from (if such a place exists), which may well be macOS itself, looking at its general preferences dialog (although I see that the colors are in fact slightly different, at least in 12.2):
>    image
> 2. New icons aren't size-optimized which I think we did at some point because the size difference in dramatic, e.g. for 2x variants, existing ones are around 275 bytes while new ones are around 2300-2600 bytes.
> 3. New icons need to be added to macosx/CMakeLists.txt: https://github.com/transmission/transmission/blob/bd895b50526665c4c379a5670105cafe017d2ad1/macosx/CMakeLists.txt#L238

This is a stop-gap PR that includes my own graphic design limitations. Each of the new images are taken from the images that existed within the codebase with personal limitations in changes until i can figure a better way; but at the scale they are, without that level of magnification, they work fine.

I would love to be able to add MacOS native icon colours in a similar fashion to what was done [here](https://github.com/transmission/transmission/blob/main/macosx/Images/TransmissionDocument.icns), but that's another PR for another day.

File: .github/workflows/objective-c-xcode.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ckerr) [reply .github/workflows/objective-c-xcode.yml:1]: Please remove this change since it's off-topic to this PR :slightly_smiling_face:

**Review by ckerr — CHANGES_REQUESTED** [review]
#2813 and #2814 should be folded into this PR. They feel like they're part of this PR rather than standalone changes.

File: .github/workflows/objective-c-xcode.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (GScottElshaw) [reply .github/workflows/objective-c-xcode.yml]: Done

**Comment by GScottElshaw** [comment]
> #2813 and #2814 should be folded into this PR. They feel like they're part of this PR rather than standalone changes.

Sorry Charles, I don't know how to do that.

**Comment by mikedld** [comment]
@GScottElshaw in my experience it should be quite easy to take existing dot image in one color and change it into another color by simply playing with various [color values](https://en.wikipedia.org/wiki/HSL_and_HSV) in your favorite image editor. You don't need to paint a single pixel yourself. You may even be able to [use ImageMagick](https://legacy.imagemagick.org/Usage/color_mods/) for that if you're more of a CLI guy.

**Comment by GScottElshaw** [comment]
> @GScottElshaw in my experience it should be quite easy to take existing dot image in one color and change it into another color by simply playing with various [color values](https://en.wikipedia.org/wiki/HSL_and_HSV) in your favorite image editor. You don't need to paint a single pixel yourself. You may even be able to [use ImageMagick](https://legacy.imagemagick.org/Usage/color_mods/) for that if you're more of a CLI guy.

You'd think, right? I'll have another go, @mikedld, but image manipulation, obviously, isn't my forte.

**Comment by GScottElshaw** [comment]
Are you happy for this to be pushed, @mikedld?

**Comment by GScottElshaw** [comment]
Can someone please review? Lots and lots of PR's piling up, from others too - a good thing! @ckerr @mikedld @livings124

File: macosx/MessageWindowController.mm [hunk 2]
```diff
@@ -268,13 +268,17 @@ - (id)tableView:(NSTableView*)tableView objectValueForTableColumn:(NSTableColumn
...
 
         case TR_LOG_INFO:
-            return [NSImage imageNamed:@"YellowDotFlat"];
+            return [NSImage imageNamed:@"GreenDotFlat"];
 
         case TR_LOG_DEBUG:
+            return [NSImage imageNamed:@"BlueDotFlat"];
+        
         case TR_LOG_TRACE:
             return [NSImage imageNamed:@"PurpleDotFlat"];
```
[the remark this answers is not in the document]
Comment (GScottElshaw) [reply macosx/MessageWindowController.mm:283]: Thanks Mike, great catch!

**Review by mikedld — APPROVED** [review]

**Comment by GScottElshaw** [comment]
@ckerr Pull the trigger please Charles.

**Comment by ckerr** [comment]
@GScottElshaw is this still of a piece with #2813 and #2814 ?

**Comment by GScottElshaw** [comment]
> @GScottElshaw is this still of a piece with #2813 and #2814 ?

@ckerr Unless i accidentally did the right thing, yes.

**Comment by GScottElshaw** [comment]
Some weirdness in the values that have nothing to do with my proposed changes. Why "yoL-9H-kTt" and "2hu-LL-6A5"?

<font key="font" metaFont="menu"/>
                            <menu key="menu" title="OtherViews" id="16">
                                <items>
                                    <menuItem title="Error" state="on" image="RedDotFlat" id="17"/>
                                    <menuItem title="Warn" image="RedDotFlat" id="yoL-9H-kTt"/>
                                    <menuItem title="Info" image="GreenDotFlat" id="18"/>
                                    <menuItem title="Debug" image="BlueDotFlat" id="19"/>
                                    <menuItem title="Trace" image="PurpleDotFlat" id="2hu-LL-6A5"/>
                                </items>
                            </menu>

**Review by ckerr — APPROVED** [review]

**Comment by GScottElshaw** [comment]
@ckerr Hi Charles, i'm sure you've seen TC isn't so happy with its Gentoo boxes right now. Do you think it's time to merge this, regardless?

**Comment by ckerr** [comment]
CI failure is unrelated to PR

**Comment by GScottElshaw** [comment]
Thank you Charles!

**Pull request merged**